mothernet/priors/classification_adapter.py

```diff
@@ -148,7 +149,7 @@ def __call__(self, batch_size, n_samples, num_features, device, epoch=None, sing
                 if random.random() < p:
```

Comment:
We could do that more simply by doing the batching outside of this prior. My concern was that this operation is a bottle neck and I'm afraid it would become much slower. Also I didn't expect a lot of improvements from sampling the categorical variables per dataset. Though that's just a guess. Do you want to do the profiling?